src/ring_theory/polynomial_algebra.lean

```diff
@@ -0,0 +1,342 @@
+    congr, simp only [linear_map.coe_mk],
+    unfold to_fun finsupp.sum monomial,
+    simp_rw [finset.smul_sum, finsupp.smul_single,  ← algebra.smul_mul_assoc],
+    refl },
+  map_add' := by {
+    intros, unfold to_fun_linear_right,
+    congr, simp only [linear_map.coe_mk],
+    unfold to_fun finsupp.sum monomial,
+    simp_rw [← finset.sum_add_distrib, ← finsupp.single_add, ← add_mul],
+    refl } }
+
+/--
+(Implementation detail).
+The function underlying `A ⊗[R] polynomial R →ₐ[R] polynomial A`,
+as a linear map.
+-/
+def to_fun_linear : A ⊗[R] polynomial R →ₗ[R] polynomial A :=
+tensor_product.lift (to_fun_bilinear R A)
+
+-- We apparently need to provide the decidable instance here
+-- in order to successfully rewrite by this lemma.
+lemma to_fun_linear_mul_tmul_mul_aux_1
+  (p : polynomial R) (k : ℕ) (h : decidable (¬p.coeff k = 0)) (a : A) :
+  ite (¬coeff p k = 0) (a * (algebra_map R A) (coeff p k)) 0 = a * (algebra_map R A) (coeff p k) :=
+begin
+  haveI := h,
+  by_cases w : ¬p.coeff k = 0,
+  { simp [w], },
+  { simp [classical.not_not.1 w], }
+end
+
+lemma to_fun_linear_mul_tmul_mul_aux_2 (k : ℕ) (a₁ a₂ : A) (p₁ p₂ : polynomial R) :
+  a₁ * a₂ * (algebra_map R A) ((p₁ * p₂).coeff k) =
+    (finset.nat.antidiagonal k).sum
+      (λ x, a₁ * (algebra_map R A) (coeff p₁ x.1) * (a₂ * (algebra_map R A) (coeff p₂ x.2))) :=
+begin
+  simp_rw [mul_assoc, algebra.commutes, ←finset.mul_sum, mul_assoc, ←finset.mul_sum],
+  congr,
+  simp_rw [algebra.commutes (coeff p₂ _), coeff_mul, ring_hom.map_sum, ring_hom.map_mul],
+end
+
+lemma to_fun_linear_mul_tmul_mul (a₁ a₂ : A) (p₁ p₂ : polynomial R) :
+  (to_fun_linear R A) ((a₁ * a₂) ⊗ₜ[R] p₁ * p₂) =
+    (to_fun_linear R A) (a₁ ⊗ₜ[R] p₁) * (to_fun_linear R A) (a₂ ⊗ₜ[R] p₂) :=
+begin
+  dsimp [to_fun_linear],
+  simp only [lift.tmul],
+  dsimp [to_fun_bilinear, to_fun_linear_right, to_fun],
+  ext k,
+  simp_rw [coeff_sum, coeff_single, finsupp.sum,
+    finset.sum_ite_eq', finsupp.mem_support_iff, ne.def, coeff_mul, finset_sum_coeff, coeff_single,
+    finset.sum_ite_eq', finsupp.mem_support_iff, ne.def,
+    mul_ite, mul_zero, ite_mul, zero_mul, apply_eq_coeff],
+  simp_rw [ite_mul_zero_left (¬coeff p₁ _ = 0) (a₁ * (algebra_map R A) (coeff p₁ _))],
+  simp_rw [ite_mul_zero_right (¬coeff p₂ _ = 0) _ (_ * _)],
+  simp_rw [to_fun_linear_mul_tmul_mul_aux_1, to_fun_linear_mul_tmul_mul_aux_2],
+end
+
+lemma to_fun_linear_algebra_map_tmul_one (r : R) :
+  (to_fun_linear R A) ((algebra_map R A) r ⊗ₜ[R] 1) = (algebra_map R (polynomial A)) r :=
+begin
+  dsimp [to_fun_linear],
+  simp only [lift.tmul],
+  dsimp [to_fun_bilinear, to_fun_linear_right, to_fun],
+  rw [← C_1, ←monomial_zero_left, finsupp.sum_single_index],
+  { simp, refl, },
+  { simp, },
+end
+
+/--
+(Implementation detail).
+The algebra homomorphism `A ⊗[R] polynomial R →ₐ[R] polynomial A`.
+-/
+-- TODO This takes about 8 seconds to elaborate. Suggestions welcome.
+def to_fun_alg_hom : A ⊗[R] polynomial R →ₐ[R] polynomial A :=
```

Comment:
```suggestion
def to_fun_alg_hom : A ⊗[R] polynomial R →ₐ[R] polynomial A :=
by exact
```
This shaves 1.5s off for me.

Reply:
But see leanprover-community/lean#383 for a better solution that will subsume this hack.

Reply:
Okay. I'm not going to merge the `by exact` change and we'll wait for the fix.